packages/generator/src/edmx-to-vdm/common/action-function-return-types.ts

```diff
@@ -150,15 +150,26 @@ function getEdmReturnType(
 
 function getEntityReturnType(
   isCollection: boolean,
-  entity: VdmEntity
+  entities: VdmEntity[]
 ): VdmFunctionImportReturnType {
-  return {
-    returnTypeCategory: VdmReturnTypeCategory.ENTITY,
-    returnType: entity.className,
-    builderFunction: entity.className,
-    isMulti: isCollection,
-    isCollection
-  };
+  if (!entities.length) {
+    throw Error('Cannot get entity return type from an empty collection.');
+  }
+
+  return entities.length === 1
+    ? {
+        returnTypeCategory: VdmReturnTypeCategory.ENTITY,
+        returnType: first(entities)!.className,
+        builderFunction: first(entities)!.className,
+        isMulti: isCollection,
+        isCollection
+      }
+    : {
+        returnTypeCategory: VdmReturnTypeCategory.ENTITY_NOT_DESERIALIZABLE,
+        returnType: 'void',
+        isMulti: isCollection,
+        isCollection
+      };
```

Comment:
I am still not a fan of `VdmReturnTypeCategory.ENTITY_NOT_DESERIALIZABLE` because strictly speaking this is not a return type category.

I am not sure that this would be an improvement, but have you considered the following options and could they make things simpler down the line?

Option 1:
Could we just return undefined here, and use this information down the line?

Option 2:
Not sure if this would even work, but could we use something like this:
```ts
{
  returnTypeCategory: VdmReturnTypeCategory.NEVER,
  returnType: 'never',
  isMulti: false,
  isCollection: false
 };
```